docarray/base_document/document.py

```diff
@@ -213,3 +215,128 @@ def _group_fields(doc: 'BaseDocument') -> _FieldGroups:
             elif dict1 is not None and dict2 is not None:
                 dict1.update(dict2)
                 setattr(self, field, dict1)
+
+    @classmethod
+    def smart_parse_obj(
+        cls: Type[T],
+        obj: Mapping[str, Any],
+        cast_map: Optional[Mapping[str, str]] = None,
+    ) -> T:
+        """
+        parse the given object, validate the data and return a new initialized objet
+        the difference from `parse_obj` is that it will smartly resolve conflict and
+        undefined key.
+
+        The `cast_map` argument is for you to define an explicit casting from the schema
+        of this Document to your object input
```

Comment:
Maybe this should be further down? At this point of the docstring I am not sure if I understand this.

Reply:
done